Hiding the GUI (or: rewriting tray logic)

Linux user here. I use a tiling WM with no system tray, and no concept of "minimizing" windows.

Typically, if a program supports hide-to-tray, I can just close the window and the program will still be running, just "closed to tray". The fact that I don't have a tray doesn't actually matter: I restore the program by re-running it. Plain and simple.

Mumble, unfortunately, doesn't have a "close to tray", but rather "minimize to tray". I have two problems with this implementation:

1) Since I cannot minimize windows in my Linux WM, I cannot hide mumble's GUI.
2) Even on Windows, this is atypical behavior. "Close to tray" is conventional, "minimize to tray" is not. To be fair, I used to see one or two programs "minimize to tray" back in the days of Windows XP, but now mumble is the one outlier that still does it. It fell out of favor because it doesn't make sense: it limits functionality. For WMs that DO support minimization, "minimize to tray" forces you to sacrifice the ability to minimize- and, having your minimized window vanish is somewhat unintuitive/non-sequitor. With "Close to tray", you can still minimize the window normally if you wish, AND you can hide to tray. Having the window vanish when you close it is intuitive. Quitting the program can be done by `File>Quit`, `Rightclick TrayIcon>Quit`, or the assigned quit hotkey.

Sorry for the wall of text, just verbosely making my case that this isn't _just_ a change for my own unique use case, but rather, this is a general design considation.

That being said, if the devs are unwilling to make the above switch, I would still like a way to hide mumble on Linux. Could I get a `--headless` commandline switch? This approach is generally less effective because I imagine there will be no way to invoke the GUI if it has been started headless....

If you do implement "close to tray", an option / commandline switch to "start hidden" would be nice, too :)

Thanks!

Building your branch fails for me with

```
AudioInput.cpp: In member function ‘int AudioInput::encodeCELTFrame(short int*, AudioInput::EncodingOutputBuffer&)’:
AudioInput.cpp:690:69: error: ‘CELT_SET_VBR_RATE’ was not declared in this scope
  cCodec->celt_encoder_ctl(ceEncoder, CELT_SET_VBR_RATE(iAudioQuality));
                                                                     ^
make[2]: *** [Makefile.Release:2816: release/AudioInput.o] Error 1
make[2]: Leaving directory '/home/Link/mumble/src/mumble'
make[1]: *** [Makefile:38: release] Error 2
make[1]: Leaving directory '/home/Link/mumble/src/mumble'
make: *** [Makefile:135: sub-src-mumble-make_first-ordered] Error 2
```

--- a/src/mumble/MainWindow.cpp
+++ b/src/mumble/MainWindow.cpp
@@ -1216,6 +1216,7 @@ void MainWindow::on_qmServer_aboutToShow() {
 	qmServer->addAction(qaServerUserList);
 	qmServer->addAction(qaServerBanList);
 	qmServer->addSeparator();
+	qmServer->addAction(qaHide);
 	qmServer->addAction(qaQuit);
 
 	qaServerBanList->setEnabled(g.pPermissions & (ChanACL::Ban | ChanACL::Write));
@@ -1791,6 +1792,10 @@ void MainWindow::on_qaUserInformation_triggered() {
 		return;
 
 	g.sh->requestUserStats(p->uiSession, false);
+}
+
+void MainWindow::on_qaHide_triggered() {
+	hide();
 }
 
 void MainWindow::on_qaQuit_triggered() {

--- a/src/mumble/MainWindow.h
+++ b/src/mumble/MainWindow.h
@@ -239,6 +239,7 @@ class MainWindow : public QMainWindow, public MessageHandler, public Ui::MainWin
 		void on_qaHelpAboutQt_triggered();
 		void on_qaHelpVersionCheck_triggered();
 		void on_qaQuit_triggered();
+		void on_qaHide_triggered();
 		void on_qteChat_tabPressed();
 		void on_qteChat_backtabPressed();
 		void on_qteChat_ctrlSpacePressed();

--- a/src/mumble/MainWindow.ui
+++ b/src/mumble/MainWindow.ui
@@ -207,6 +207,17 @@
     <string>Ctrl+Q</string>
    </property>
   </action>
+  <action name="qaHide">
+   <property name="text">
+    <string>&amp;Hide Mumble</string>
+   </property>
+   <property name="toolTip">
+    <string>Hides the main Mumble window.</string>
+   </property>
+   <property name="whatsThis">
+    <string>Hides the main Mumble window. Restore by clicking on the tray icon or starting Mumble again.</string>
+   </property>
+  </action>
   <action name="qaServerConnect">
    <property name="icon">
     <iconset>

